src/spikeinterface/preprocessing/common_reference.py

```diff
@@ -1,25 +1,47 @@
+from __future__ import annotations
 import numpy as np
+from typing import Optional, Literal
 
 from spikeinterface.core.core_tools import define_function_from_class
 
 from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
 from ..core import get_closest_channels
+from spikeinterface.core.baserecording import BaseRecording
 
 from .filter import fix_dtype
 
 
 class CommonReferenceRecording(BasePreprocessor):
     """
-    Re-references the recording extractor traces.
+    Re-references the recording extractor traces. That is, the value of the traces are
+    shifted so the there is a new zero (reference).
+
+    The new reference can be estimated either by using common median reference (CMR) or
+    common average reference (CAR).
+
+    The new reference can be set three ways:
+        * "global": the median/average of all channels is set as the new reference.
+        That is, the median/average is subtracted from all channels.
+        * "single": In the simplest case, a single channel from the recording is set as the new reference.
+        That is, the channel is subtracted from all other channels. To use this option, the `ref_channel_ids` argument
+        is used with a single channel id. Note that this option will zero out the reference channel.
+        A collection of channels can also be used as the new reference. In this case, the median/average of the
+        selected channels is subtracted from all other channels. To use this option, pass the group of channels as
+        a list in `ref_channel_ids`.
+        * "local": the median/average within an annulus is set as the new reference. The parameters
+        of the annulus are specified using the `local_radius` argument. With this option, both close channels and
+        far channels are excluded from the median/average. Note that setting the `local_radius` to (0, exclude_radius)
+        will yield a simple circular local region.
+
 
     Parameters
     ----------
     recording: RecordingExtractor
         The recording extractor to be re-referenced
     reference: "global" | "single" | "local", default: "global"
-        If "global" then CMR/CAR is used either by groups or all channel way.
-        If "single", the selected channel(s) is remove from all channels. operator is no used in that case.
-        If "local", an average CMR/CAR is implemented with only k channels selected the nearest outside of a radius around each channel
+        If "global" the reference is the average or mean across all the channels.
+        If "single", the reference is a single channel or a list of them that need to be set with the `ref_channel_ids`.
```

Comment:
```suggestion
        If "single", the reference is a single channel or a list of channels that need to be set with the `ref_channel_ids`.
```
explicit better than implicit no? :)

Reply:
Definitley! In writing avoid pronouns as much as you can.